arangod/Cluster/ParticipantsCacheFeature.h

```diff
@@ -0,0 +1,63 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2022 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Alex Petenchea
+/// @author Manuel Pöter
+////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include "RestServer/arangod.h"
+#include "Scheduler/Scheduler.h"
+
+#include <memory>
+#include <unordered_map>
+
+namespace arangodb::cluster {
+struct IFailureOracle;
+class ParticipantsCache;
+
+class ParticipantsCacheFeature final
+    : public ArangodFeature,
```

Comment:
I dislike the name `ParticipantsCache`. Is is very board could be anything and the thing it does is the last thing I think about given the name, `Participant` is a term, that is exclusively used in the replicated logs business and in all other places these are referred to as servers. And it's not the participants that are cached. 

`ServerFailureCache` would be better. Or `FailureOracleFeature` and than `ParticipantsCache` becomes simply `FailureOracleImpl` 🤷🏽 names are the hardest but most important part ...

Reply:
Indeed, the name was a generic one. I was thinking it's going to have some other functionality in the future, but it's worth renaming it later and having a more specific name right now. I'm using `FailureOracleFeature` and `FailureOracleImpl`.